### PR TITLE
Tcti cleanup and state machine docs

### PR DESCRIPTION
**Pull request by flihp**
This should be the last round of cleanup commits to bring the TCTI stuff in-line with our coding standard & conventions. The second half of this PR starts meddling around with the internals of the TCTI context structure and the state machine that tracks the transmit / receive stuff.

### CODE REVIEW

File: tcti/tcti.h [hunk 1]
```diff
@@ -69,12 +69,42 @@ typedef struct {
...
+ *   RECEIVE:
+ *     transmit:    produces TSS2_TCTI_RC_BAD_SEQUENCE
+ *     receive:     success transitions the state machine to TRANSMIT
+ *                  failure with the following RCs leave the state unchanged:
+ *                    TRY_AGAIN, INSUFFICIENT_BUFFER, BAD_CONTEXT,
+ *                    BAD_REFERENCE, BAD_VALUE, BAD_SEQUENCE
+ *                  all other failures transition state machine to
+ *                    TRANSMIT (not recoverable)
+ *     finalize:    transitions state machine to FINAL state
+ *     cancel:      success transitions state machine to READY_TRANSMIT
```
Comment (tstruk) [inline tcti/tcti.h]: Where is the READY_TRANSMIT defined? or is it just STATE_TRANSMIT?

File: tcti/tcti_socket.c [hunk 2]
```diff
@@ -191,14 +191,22 @@ tcti_socket_cancel (
     TSS2_TCTI_CONTEXT_INTEL *tcti_intel = tcti_context_intel_cast (tctiContext);
     TSS2_RC rc;
 
+    /* the TCTI must have executed the transmit function successfully */
     rc = tcti_common_checks (tctiContext);
     if (rc != TSS2_RC_SUCCESS) {
         return rc;
-    } else if (tcti_intel->status.commandSent != 1) {
+    }
+    if (tcti_intel->state != TCTI_STATE_RECEIVE) {
```
Comment (tstruk) [inline tcti/tcti_socket.c]: It look like now the status.commandSent is only used in setLocality and we can use the tcti_intel->state and remove the status.commandSent.

File: tcti/tcti.h (hunk 1 above)
Comment (flihp) [reply tcti/tcti.h]: That's an old name for the `TRANSMIT` state (aka `TCTI_STATE_TRANSMIT`). Will fix this up.

**Comment by tstruk** [comment]
First three commits look ok. Submitted some comments on the fourth one.

File: tcti/tcti_socket.c (hunk 2 above)
Comment (flihp) [reply tcti/tcti_socket.c]: I wrote another series today that removes all of the `status.*` stuff. Will make another PR once this goes in.

File: tcti/tcti.h [hunk 3]
```diff
@@ -69,20 +69,42 @@ typedef struct {
...
-
-enum tctiStates { TCTI_STAGE_INITIALIZE, TCTI_STAGE_SEND_COMMAND, TCTI_STAGE_RECEIVE_RESPONSE };
+/*
+ * The elements in this enumeration represent the possible states that the
+ * TCTI can be in. The state machine is as follows:
+ * An instantiated TCTI context begins in the TRANSMIT state:
+ *   TRANSMIT:
+ *     transmit:    success transitions the state machine to RECEIVE
+ *                  failure leaves the state unchanged
+ *     receieve:    produces TSS2_TCTI_RC_BAD_SEQUENCE
```
Comment (dantpm) [inline tcti/tcti.h]: typo: s/receieve/receive/

**Comment by flihp** [comment]
This is the second time I've seen `test/integration/sys-initialize.int` fail this week but I can't repro reliably.

**Comment by flihp** [comment]
hu. maybe clang

**Comment by flihp** [comment]
my fault, not sure how but I dropped a commit that fixed this

File: test/integration/sys-initialize.int.c [hunk 4]
```diff
@@ -44,8 +44,8 @@ test_invoke (TSS2_SYS_CONTEXT *sapi_context)
     }
 
     // NOTE: don't do this in real applications.
-    tctiContextIntel.transmit = (TSS2_TCTI_TRANSMIT_FCN)0;
-    tctiContextIntel.receive = (TSS2_TCTI_RECEIVE_FCN)1;
+    TSS2_TCTI_RECEIVE (sapi_context) = (TSS2_TCTI_RECEIVE_FCN)1;
```
Comment (flihp) [inline test/integration/sys-initialize.int.c]: this is why the clang build is b0rked

**Comment by flihp** [comment]
All comments addressed and got that failing test case worked out. Merging ...

**Pull request merged**